python/feathub/processors/flink/table_builder/redis_utils.py

```diff
@@ -96,9 +103,60 @@ def append_physical_key_columns_per_feature(
     return table, physical_key_column_names
 
 
+def optimize_redis_source_lookup_map(
+    t_env: StreamTableEnvironment,
+    table: NativeFlinkTable,
+    table_descriptor: TableDescriptor,
+    join_field_descriptors: Dict[str, JoinFieldDescriptor],
+) -> NativeFlinkTable:
+    """
+    If the right table of a lookup join is a RedisSource with a single map-typed
+    feature, and only designated keys of the map is used in the join, the Redis
+    source table would be optimized to only lookup the designated fields in the
+    Redis hash, instead of reading all hash fields.
+    """
+    if not isinstance(table_descriptor, RedisSource):
+        return table
+
+    feature_names = [
+        x for x in table_descriptor.schema.field_names if x not in table_descriptor.keys
+    ]
+
+    if len(feature_names) != 1 or not isinstance(
```

Comment:
Is there particular reason that this optimization only support RedisSource with one field?